### PR TITLE
feat: rename metadata file

### PR DESCRIPTION
**Pull request by zdtsw**
- previous named as *sbom<XXX>.json
	- now change to *sbom<XXX>-metadata.json
	- update filter on binary files to include json (sbom<XXX>.json
	- this could solve problem with *sbom*.json file which does not generate metadata any more
	- this could help "api" to filter sbom's  metadata
Ref: https://github.com/adoptium/ci-jenkins-pipelines/issues/328

### CODE REVIEW

**Comment by zdtsw** [comment]
How you feel about this PR? @smlambert @andrew-m-leonard 
Could rename break anything else? 
some discussion https://adoptium.slack.com/archives/CFUJV9XV0/p1655367001333359?thread_ts=1653911822.234059&cid=CFUJV9XV0

**Comment by andrew-m-leonard** [comment]
@zdtsw the sbom and the old metadata file are two different things, are we sure we are not confusing things here?

**Comment by zdtsw** [comment]
> @zdtsw the sbom and the old metadata file are two different things, are we sure we are not confusing things here?

so I think in the past we only create *.tar.gz or *.zip/pkg etc file, and then based on these filename, we create *.json (which is the metadata)
then we added support for sbom, which is named as sbom*.json, so the function does not create metadata for the sbom file.
so the api part is not getting sbom's metadata file.

what I am thinking is to, keep sbom as the sbom*.json file, but make all the metadata json file from "*.json" to  "*-metadata.json"
So in case of sbom, it should have a sbom*.json and its metadata sbom*-metadata.json

Then in API, we only filter out *-metadata.json

**Comment by andrew-m-leonard** [comment]
It would be good to see the User scenario that has a problem, for example API query to get SBOM is failing: ....exception...?

**Comment by zdtsw** [comment]
> It would be good to see the User scenario that has a problem, for example API query to get SBOM is failing: ....exception...?

The current problem is, we do not generate metadata file for sbom. 
that is causing api service cannot download sbom json nor get assets of it (because of missing metadata on sbom)


e.g https://api.adoptium.net/v3/binary/latest/18/ga/linux/x64/sbom/hotspot/normal/eclipse?project=jdk
return 404 Error: Not Found
e.g https://api.adoptium.net/v3/assets/latest/18/hotspot?architecture=x64&image_type=sbom&os=linux&vendor=eclipse
return 200 with [] body

**Comment by andrew-m-leonard** [comment]
@zdtsw ok lets discuss as I don't really want a second json file for the sbom.json, seems unnecessary, lets try and fix the API server so it doesn't need it... surely it can just serve up the sbom.json....?
Seems a bit silly to effectively have artifacts for:
```
sbom.json
sbom-metadata.json
```

**Comment by zdtsw** [comment]
If I no mistaken, we need to add all these fields into sbom.json in order for it to be severed as a metadata 
https://github.com/adoptium/api.adoptium.net/blob/main/adoptium-updater-parent/adoptium-datasources-parent/adoptium-github-datasource/src/main/kotlin/net/adoptium/api/v3/dataSources/github/graphql/models/GHMetaData.kt#L30
(warning is not a mandatory filed)

**Comment by andrew-m-leonard** [comment]
Ah right, starting to understand the problem, let me have a think

**Comment by zdtsw** [comment]
After discussion:
only add -metadata into sbom's metadata file.

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -1050,6 +1050,10 @@ class Build {
                 metaWrittenOut = true
             }
 
+            // Special handling for sbom metadat file (to be backwards compatible for api service)
+            if (file.contains("sbom")) {
+                file.replace(".json", "-metadata") // from "*sbom<XXX>.json" to "*sbom<XXX>-metadata"
```
Comment (andrew-m-leonard) [inline pipelines/build/common/openjdk_build_pipeline.groovy:1055]: not sure you want to inline replace file variable, I would put the context.writeFile line in both sides of an if (file.contains("sbom")) {context.writeFile file:file.replace(".json", "-metadata") .....} else {..as before.}

**Review by andrew-m-leonard — APPROVED** [review]
looks good

**Comment by zdtsw** [comment]
@smlambert if you have time, could you do a review on it and if good to merge it? want to have it tested out and see if it works with api, thanks!

**Review by smlambert — APPROVED** [review]
thanks @zdtsw !

**Pull request merged**